### PR TITLE
arelle: 2017-06-01 -> 2017-08-24

### PR DESCRIPTION
**Pull request by disassembler**
###### Motivation for this change
fixes tests for python 3.6.

Remove \m from regex (not supported anymore)

Also, implicitly specify from arelle import in Cntlr in CntlrProfiler.py

arelle released a new release for RedHat on 2017-08-24, so I'm considering code from that date a stable release.

tests break on headless expecting tkinter library, so tests are only ran on when gui == true.

related to #28643

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by disassembler** [comment]
@roberth can I get a review since your the maintainer?

File: pkgs/development/python-modules/arelle/tests.patch [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+diff --git c/arelle/ValidateFilingText.py i/arelle/ValidateFilingText.py
+index 12dbbbb..c0e98c3 100644
+--- c/arelle/ValidateFilingText.py
++++ i/arelle/ValidateFilingText.py
+@@ -16,7 +16,7 @@ XMLdeclaration = re.compile(r"<\?xml.*\?>", re.DOTALL)
+ XMLpattern = re.compile(r".*(<|&lt;|&#x3C;|&#60;)[A-Za-z_]+[A-Za-z0-9_:]*[^>]*(/>|>|&gt;|/&gt;).*", re.DOTALL)
+ CDATApattern = re.compile(r"<!\[CDATA\[(.+)\]\]")
+ #EFM table 5-1 and all &xxx; patterns
+-docCheckPattern = re.compile(r"&\w+;|[^0-9A-Za-z`~!@#$%&\*\(\)\.\-+ \[\]\{\}\|\\:;\"'<>,_?/=\t\n\r\m\f]") # won't match &#nnn;
++docCheckPattern = re.compile(r"&\w+;|[^0-9A-Za-z`~!@#$%&\*\(\)\.\-+ \[\]\{\}\|\\:;\"'<>,_?/=\t\n\r\f]") # won't match &#nnn;
```
Comment (roberth) [inline pkgs/development/python-modules/arelle/tests.patch]: I had to look this up. The table mentioned in the comment ( EFM table 5-1 in https://www.sec.gov/info/edgar/edgarfm-vol2-v25.pdf ) does not seem to include anything that warrants the `\m`. Also it's hard to find a meaning for `\m` in a python regex. Probably ok. I'll check with the author asynchronously.

**Comment by roberth** [comment]
@disassembler, thank you for looking into this. Removing the `\m` seems to be the right thing to do. I'll check with the author to be sure, but the spec seems to suggest it is ok. Looks good to me, but I can't merge it because github doesn't know I'm the maintainer. Calling @chaoflow

File: pkgs/development/python-modules/arelle/default.nix [hunk 2]
```diff
@@ -46,6 +49,8 @@ buildPythonPackage {
     (cd apidocs && make html && cp -r _build $doc)
     '';
 
+  doCheck = if gui then true else false;
+
```
Comment (Mic92) [inline pkgs/development/python-modules/arelle/default.nix]: I think python2 is no longer supported. `./bin/arelle` is python3-only and https://github.com/Arelle/Arelle/blob/master/setup.py#L424 only lists python3

**Pull request merged**